### PR TITLE
Added VelocityVectorProperty

### PR DESCRIPTION
**Pull request by tfili**
Added `VelocityVectorProperty` so billboard's aligned axis can follow the velocity vector. Tests were also added.

### CODE REVIEW

**Comment by pjcozzi** [comment]
@hpinkos can you review this?

File: Source/DataSources/VelocityVectorProperty.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        }
+        //>>includeEnd('debug');
+
+        if (!defined(result)) {
+            result = new Cartesian3();
+        }
+
+        var property = this._position;
+        if (Property.isConstant(property)) {
+            return undefined;
```
Comment (hpinkos) [inline Source/DataSources/VelocityVectorProperty.js]: Wouldn't this be the zero vector instead of undefined?

**Comment by mramato** [comment]
@tfili can we rewrite `VelocityOrientationProperty` in terms of this now?  I imagine it would get rid of all of the code duplication.

File: Specs/DataSources/VelocityVectorPropertySpec.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    'use strict';
+
+    var time = JulianDate.now();
+
+    it('can default construct', function() {
+        var property = new VelocityVectorProperty();
+        expect(property.isConstant).toBe(true);
+        expect(property.definitionChanged).toBeInstanceOf(Event);
+        expect(property.position).toBeUndefined();
+        expect(property.getValue(time)).toBe(undefined);
```
Comment (hpinkos) [inline Specs/DataSources/VelocityVectorPropertySpec.js]: `.toBe(undefined)` -> `.toBeUndefined()`

File: Specs/DataSources/VelocityVectorPropertySpec.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    });
+
+    it('can construct with arguments', function() {
+        var position = new SampledPositionProperty();
+        var property = new VelocityVectorProperty(position);
+        expect(property.isConstant).toBe(true);
+        expect(property.definitionChanged).toBeInstanceOf(Event);
+        expect(property.position).toBe(position);
+        expect(property.getValue(time)).toBe(undefined);
```
Comment (hpinkos) [inline Specs/DataSources/VelocityVectorPropertySpec.js]: same as above

**Comment by hpinkos** [comment]
@tfili just had that one question

File: Source/DataSources/VelocityVectorProperty.js (hunk 1 above)
Comment (hpinkos) [inline Source/DataSources/VelocityVectorProperty.js]: Well, I guess if you wanted the vector to always be normalized this makes sense.  Just wanted to double check.

Comment (tfili) [inline Source/DataSources/VelocityVectorProperty.js]: The other properties seem to return undefined when there isn't a value at that time. This code was pretty much copied from the `VectorOrientationProperty`. @mramato is returning `undefined` the right thing to do here?

**Comment by tfili** [comment]
@mramato this should be good for another look.

File: Source/DataSources/VelocityVectorProperty.js (hunk 1 above)
Comment (mramato) [inline Source/DataSources/VelocityVectorProperty.js]: Yes, `undefined` just gets translated to "use the default" which is probably the best case scenario here.

File: Source/DataSources/VelocityVectorProperty.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+     * A {@link Property} which evaluates to a {@link Cartesian3} vector
+     * based on the velocity of the provided {@link PositionProperty}.
+     *
+     * @alias VelocityVectorProperty
+     * @constructor
+     *
+     * @param {Property} [position] The position property used to compute the velocity.
+     *
+     * @example
+     * //Create an entity with position and orientation.
```
Comment (mramato) [inline Source/DataSources/VelocityVectorProperty.js]: This example is now incorrect.  I would recommend updating it to illustrate orienting a billboard along it's velocity vector.

**Comment by mramato** [comment]
Just that one last comment.

**Comment by tfili** [comment]
@mramato Updated the code sample.

File: Source/DataSources/VelocityVectorProperty.js (hunk 4 above)
Comment (mramato) [inline Source/DataSources/VelocityVectorProperty.js]: The comment is still incorrect.  Change it to something like `//Create an entity with a billboard rotated to match its velocity`

**Comment by mramato** [comment]
Also, please merge in master.

**Comment by tfili** [comment]
@mramato Master is merged and the comment is fixed.

**Comment by mramato** [comment]
Thanks @tfili

**Pull request merged**